### PR TITLE
Update `is_frontend_url()` to verify passed url path with `home_url()`

### LINKED ISSUE #7329: wp_nav_menu suffix in Custom links to remote websites
### Bug Description

This plugin adds a suffix /amp to my custom links to remote websites in my footer menu (wp_nav_menu).

My blog is under a subpath (/blog) so I was able to manually remove this suffix (and keep it only for my blog links) with :

```
		if(strstr($parsed_url['path'], 'blog')){
			$parsed_url['path'] .= user_trailingslashit( amp_get_slug(), 'amp' );
		}
```
in `amp/src/PairedUrl.php`.

### Expected Behaviour

The suffix must no be added to custom links in wp menus.

### Screenshots

_No response_

### PHP Version

7.3

### Plugin Version

latest

### AMP plugin template mode

Transitional

### WordPress Version

_No response_

### Site Health

_No response_

### Gutenberg Version

_No response_

### OS(s) Affected

_No response_

### Browser(s) Affected

_No response_

### Device(s) Affected

_No response_

### Acceptance Criteria

_No response_

### Implementation Brief

_No response_

### QA Testing Instructions

_No response_

### Demo

_No response_

### Changelog Entry

_No response_

### PR DESCRIPTION
**Pull request by thelovekesh**
## Summary
Apart from fixing #7329 this PR also fixes the following things:
- Errors reported by the new PHPStan version(v1.9.2).
- Failing E2E tests. These errors were caused due to missing `twentytwenty` theme from the WP distribution.
- Failing PHPUnit tests due to the new Gutenberg release.

Fixes #7329 

## Checklist

- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: includes/sanitizers/class-amp-link-sanitizer.php [hunk 1]
```diff
@@ -79,7 +86,9 @@ public function __construct( $dom, array $args = [] ) {
 
 		parent::__construct( $dom, $args );
 
-		$this->home_host    = wp_parse_url( home_url(), PHP_URL_HOST );
+		$parsed_home        = wp_parse_url( home_url() );
+		$this->home_host    = $parsed_home['host'];
+		$this->home_path    = $parsed_home['path'];
```
Comment (westonruter) [inline includes/sanitizers/class-amp-link-sanitizer.php]: Just to be safe:

```suggestion
		$parsed_home        = wp_parse_url( home_url( '/' ) );
		$this->home_host    = $parsed_home['host'] ?? null;
		$this->home_path    = $parsed_home['path'] ?? '/';
```

File: includes/sanitizers/class-amp-link-sanitizer.php [hunk 2]
```diff
@@ -295,6 +304,11 @@ public function is_frontend_url( $url ) {
 			return false;
 		}
 
+		// Skip adding query var to links on other paths.
+		if ( ! empty( $parsed_url['path'] ) && false !== strpos( $parsed_url['path'], $this->home_path ) ) {
```
Comment (westonruter) [inline includes/sanitizers/class-amp-link-sanitizer.php]: This doesn't seem quite right. We need to check if the parsed path _begins_ with the home path, not that it exists anywhere in the string. Consider case where WP is installed at `/blog/` but the URL being linked to `/about/my/blog/`.

```suggestion
		if ( ! empty( $parsed_url['path'] ) && 0 !== strpos( $parsed_url['path'], $this->home_path ) ) {
```

**Review by westonruter — CHANGES_REQUESTED** [review]
Would you also please add tests to ensure this is working as expected?

File: includes/sanitizers/class-amp-link-sanitizer.php (hunk 2 above)
Comment (thelovekesh) [reply includes/sanitizers/class-amp-link-sanitizer.php]: Ahh. got it. Thanks for catching it 👍🏼 .

**Comment by thelovekesh** [comment]
@westonruter This PR has been updated to fix all static analysis, PHPUnit, and e2e test issues. I will revert related changes from other PRs as they might get delayed in merging and failing tests cause some confusion while reviewing the PRs.

**Review by westonruter — APPROVED** [review]
🎉

**Pull request merged**